tower-http/src/cors/mod.rs

```diff
@@ -682,13 +685,15 @@ where
             KindProj::CorsCall { future, headers } => {
                 let mut response: Response<B> = ready!(future.poll(cx))?;
 
+                let response_headers = response.headers_mut();
+
                 // vary header can have multiple values, don't overwrite
                 // previously-set value(s).
-                if let Some(vary) = headers.remove(header::VARY) {
+                if let Some(vary) = response_headers.remove(header::VARY) {
                     headers.append(header::VARY, vary);
```

Comment:
Aha, I see the problem now! This was supposed to be the actual response's headers. Can you revert the other changes in this fn and in the test and change this to append to `response`s headers?

Reply:
Done ✅